### PR TITLE
Fixed aligment issue for Raspberry Pi 2B

### PR DESCRIPTION
**Pull request by technotive**
The `.align 8` instruction resulted in rc12 being located an invalid constant address
Optionally `.align 2` can be used to make the offset explicit.

Can not verify impact on other devices, since I have no other devices.

### CODE REVIEW

**Comment by gvanas** [comment]
Thanks for the fix.

Can't `.align 4` work on your device? Because `. align 2` can cause performance degradation if the round constants cross 32-bit boundaries.

**Comment by technotive** [comment]
It seems that 4 does idneed not work. One fix that could work while keeping `.align 8` or `.align 4` would be to revert back to using `ldr`, but it looks like the point was to not use it.

Would you like me to try some other things/ if it is indeed (mostly) limitted to my device, I don't mind trying some alternative approaches first.

**Comment by gvanas** [comment]
Wow, this looks like a lot of work!

Could you summarize your contribution? Since there are so many commits, I have to admit I am a bit lost. Thanks!

**Comment by technotive** [comment]
Whoops, I totally forgot this one was outstanding.
I am working on optimizing Xoodoo for ARMv7-A (On the Raspberry Pi 2B), in this context, this branch now:
- Adds a fully functional ARMv7-A NEON/SIMD implementation of Xoodoo. This is a Time4 implementation.
- Currectly this implementation is slightly slower than the regular implementation.
- The Raspberry Pi 2B has issues with some of the `.align` statements (Specifically in Xoodyak, if I recall correctly). and I have found no nice working ones. I left them out in this branch.

Since the logic of Xoodoo is not tied to the offending aligns, it is perfectly fine (in my opinion) to drop that part and just merge in the Times4 Xoodoo implementation, as long as it works on other platforms. I will update my branch as needed whenever I fetch upstream.

I have not touched any code outside those aligns and the relevant Xoodoo files.

Some commits are just clumsy reverts (CTRL-Z + Commit) since I use this repo to send the code to the Raspberry. I tried to keep it clean-ish but it is kind of a mess. Sorry.

**Comment by gvanas** [comment]
Could you squash your commits? Then I will merge them.

**Comment by technotive** [comment]
This should do the trick, squashed and verified to run on the Raspberry Pi 2B.

**Comment by gvanas** [comment]
Thanks a lot for your contribution!
